lib/pbench/common/utils.py

```diff
@@ -126,11 +126,14 @@ def cleanup(self):
         This handles errors and reports them, but doesn't propagate failure to
         ensure that cleanup continues as best we can.
         """
+        self.logger.debug(f"beg - {self.name}")
         try:
             self.action()
         except Exception:
             # TODO: f-string used because this is shared by agent and server
             self.logger.exception(f"Unable to {self}")
+        else:
+            self.logger.debug(f"end - {self.name}")
 
```

Comment:
The concern about the use of an f-string at line 133 is misplaced:  that is, I'm reasonably certain that we will always log the exception, and so there is nothing to be saved by deferring the formatting (so the use of an f-string here presents no problem).

On the other hand, at lines 129 and 136, we will (presumably) _usually_ be skipping the logging of debug messages, and so the use of an f-string there is disappointing but understandable (and worth noting with a `TODO`).

(So, if there is cause to edit things for some other reason, we should remove line 133 and place it before each of lines 129 and 136....)

Reply:
The problem here is with the formatting string itself when not using an f-string.  So the comment really applies to lines 128, 134, and 136.

You would need to use `self.logger.debug("end - {}", self.name)` when it is in the server code, and `self.logger.debug("end - %s", self.name)` when it is in the agent code.

See issue #2025.

Reply:
I concur.  But, in the case of line 134, there is no benefit to deferring the formatting, so we might as well use the f-string there, because it is more intuitive for the reader.  And, therefore, there is no point to having the code comment at line 133 (so it should be removed).

On the other hand, re #2025, we _should_ have a comment like the one at line 133 before each of the new calls to `logger.debug()`, because those calls _would_ benefit enough from deferring the formatting to justify the additional complexity...once we can.